assets/tests/hooks/useAlerts.test.ts

```diff
@@ -0,0 +1,92 @@
+
+describe("useAlerts", () => {
+  test("alerts is empty to start with", () => {
+    const { result } = renderHook(() => useAlerts(undefined, []))
+    expect(result.current).toEqual({})
+  })
+
+  test("selecting a new route subscribes to the new channel", () => {
+    const mockSocket = makeMockSocket()
+    const mockChannel = makeMockChannel()
+    mockSocket.channel.mockImplementationOnce(() => mockChannel)
+
+    const { rerender } = renderHook(() => useAlerts(mockSocket, ["1"]))
+
+    // Needs to be kicked to do the effects again after the socket initializes
+    rerender()
+
+    expect(mockSocket.channel).toHaveBeenCalledTimes(1)
+    expect(mockSocket.channel).toHaveBeenCalledWith("alerts:route:1")
+    expect(mockChannel.join).toHaveBeenCalledTimes(1)
+  })
+
+  test("unselecting a route unsubscribes from the channel", () => {
+    const mockSocket = makeMockSocket()
+    const mockChannel = makeMockChannel()
+    mockSocket.channel
+      .mockImplementationOnce(() => mockChannel)
+      .mockImplementationOnce(() => mockChannel)
+      .mockImplementationOnce(() => mockChannel)
+
+    const { rerender } = renderHook(
+      (selectedRouteIds: RouteId[]) => useAlerts(mockSocket, selectedRouteIds),
+      { initialProps: ["1", "2"] }
+    )
+    rerender(["1"]) // Deselect the route
+
+    expect(mockChannel.leave).toHaveBeenCalledTimes(1)
+  })
+
+  test("returns results from joining a channel", async () => {
+    const mockSocket = makeMockSocket()
+    const mockChannel = makeMockChannel()
+    mockSocket.channel.mockImplementationOnce(() => mockChannel)
+    mockChannel.receive.mockImplementation((event, handler) => {
+      if (event === "ok") {
+        handler({ data: ["alert text"] })
+      }
+      return mockChannel
+    })
+
+    const { result } = renderHook(() => useAlerts(mockSocket, ["1"]))
+
+    expect(result.current).toEqual({
+      "1": ["alert text"],
+    })
+  })
+
+  test("returns results pushed to the channel", async () => {
+    const mockSocket = makeMockSocket()
+    const mockChannel = makeMockChannel()
+    mockSocket.channel.mockImplementationOnce(() => mockChannel)
+    mockChannel.on.mockImplementation((event, handler) => {
+      if (event === "alerts") {
+        handler({ data: ["alert text"] })
+      }
+    })
+
+    const { result } = renderHook(() => useAlerts(mockSocket, ["1"]))
+
+    expect(result.current).toEqual({
+      "1": ["alert text"],
+    })
+  })
+
+  test("reloads the window on channel timeout", async () => {
+    const reloadSpy = jest.spyOn(browser, "reload")
+    reloadSpy.mockImplementationOnce(() => ({}))
+    const mockSocket = makeMockSocket()
+    const mockChannel = makeMockChannel("timeout")
+    mockSocket.channel.mockImplementationOnce(() => mockChannel)
+
+    renderHook(() => useAlerts(mockSocket, ["1"]))
+
+    expect(reloadSpy).toHaveBeenCalled()
+    reloadSpy.mockRestore()
+  })
+})
```

Comment:
nit: I think duplicate calls to `mockImplementationOnce` like `mockSocket.channel.mockImplementationOnce(() => mockChannel)` should be grouped into one `.mockImplementation(() => mockChannel)` if possible?